src/test/ui/imports/issue-55884-2.stderr

```diff
@@ -1,8 +1,20 @@
 error[E0603]: struct `ParseOptions` is private
   --> $DIR/issue-55884-2.rs:12:17
    |
+LL |     pub struct ParseOptions {}
+   |     -------------------------- consider importing struct `::options::ParseOptions` directly
+...
+LL |     pub use options::*;
+   |             ---------- ...through this public re-export of `ParseOptions`
+...
+LL |     use ParseOptions;
+   |         ------------ used restricted re-export of `ParseOptions`...
+...
 LL | pub use parser::ParseOptions;
-   |                 ^^^^^^^^^^^^
+   |         --------^^^^^^^^^^^^
+   |         |       |
+   |         |       this struct is private
+   |         ...through this public re-export of `ParseOptions`...
```

Comment:
The weird span here is suboptimal but it seems to be a side-effect of the import resolution. I can fix this case in particular with a span overlap check, but I'd prefer to fix the underlying issue.